docs/overview.rst

```diff
@@ -1,20 +1,110 @@
 Why new programming language
 ============================
 
-Taichi is a high-performance programming language for computer graphics applications. The design goals are
+Taichi is a high-performance programming language for computer graphics applications.
+
+Introduction
+------------
+
+Different backends
+******************
+
+Before Taichi has been invented, to write a **general purposed GPU (GPGPU)**
+program, we have to manipulate through tons of different APIs, e.g.:
+
+``cuDeviceSynchronize()`` for CUDA, ``glDispatchCompute()`` for OpenGL,
+``pthread_join()`` for multi-core CPUs...
+
+Downside:
+
+1. CUDA, for example, despite its powerfulness, can run only on supported NVIDIA
+   GPUs. One could not share/distribute his/her work with/to a computer
+   without NVIDIA GPUs.
+   This means the loss of portability.
+
+2. OpenGL, due to its historical debt, is too complicated for beginners when
+   used as a GPGPU programming tool. People could easily get
+   lost in API manuals instead of focusing on their original goals.
+   This could cause the loss of productivity.
+
+3. GPU programming langurages are often less flexible and less functional.
+   It's easy to make use of high-level C++ features and external libraries
+   on CPU. But it would be hard, at least in the OpenGL shader langurage,
+   to do so.
+
+4. It's relatively hard to visualize the results in GUI. You will have to
+   make a bridge to your native GUI system before start to debug the program
+   so that you could debug with it.
+
+5. It's very hard to optimize data structures for sparse computation accessing
+   tensors / fields with plain indices. It would be great if we could decouple
+   computations from data structures.
+
+6. It is often the case that we need to compute the derivative of a kernel
+   function, e.g. compute forces from gradients of potential energy. We will
+   have to calculate the derivative by hand or external tools like Matlab.
+
```

Comment:
```suggestion

7. A lot of repetitive work during study or research can be pretty distressing. This will not only waste a lot of programming time, and kill off many beginner's interest. A high integrated tool will significantly improve the situation.
```